include/triton/core/tritonserver.h

```diff
@@ -905,6 +905,28 @@ TRITONSERVER_InferenceRequestAppendInputData(
     const void* base, size_t byte_size, TRITONSERVER_MemoryType memory_type,
     int64_t memory_type_id);
 
+/// Assign a buffer of data to an input for execution on a particular device. 
+/// The buffer will be appended to any existing buffers for that input on that device.
+/// The 'inference_request' object takes ownership of the buffer and so the caller 
+/// should not modify or free the buffer until that ownership is released by
+/// 'inference_request' being deleted or by the input being removed
+/// from 'inference_request'.
+///
+/// \param inference_request The request object.
+/// \param name The name of the input.
+/// \param base The base address of the input data.
+/// \param byte_size The size, in bytes, of the input data.
+/// \param memory_type The memory type of the input data.
+/// \param memory_type_id The memory type id of the input data.
+/// \param device_id The ID of the device that will use this input for execution.
+/// \param device_kind The type of device that will use this input for execution.
+/// \return a TRITONSERVER_Error indicating success or failure.
+TRITONSERVER_DECLSPEC TRITONSERVER_Error*
+TRITONSERVER_InferenceRequestAppendInputDataForDevice(
+    TRITONSERVER_InferenceRequest* inference_request, const char* name,
+    const void* base, size_t byte_size, TRITONSERVER_MemoryType memory_type,
+    int64_t memory_type_id, int device_id, TRITONSERVER_InstanceGroupKind device_kind);
```

Comment:
I think you need to add the id of the target device as well, so you can add different sets of the data for different devices with the same kind.